ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/viewmodel/CustomerCenterViewModel.kt

```diff
@@ -451,16 +466,18 @@ internal class CustomerCenterViewModelImpl(
     }
 
     override fun onNavigationButtonPressed() {
-        _state.update { currentState ->
-            when {
-                currentState is CustomerCenterState.Success &&
-                    currentState.navigationButtonType == CustomerCenterState.NavigationButtonType.BACK -> {
-                    currentState.resetToMainScreen()
-                }
-                currentState is CustomerCenterState.Success -> {
-                    CustomerCenterState.NotLoaded
+        if (_state.value is CustomerCenterState.Success && onNavigationOverride != null) {
+            onNavigationOverride?.invoke()
+            onNavigationOverride = null
```

Comment:
Is nulling out the variable a needed? I wasn't expecting that

Reply:
Oh got it. This gets set when the promotional offer is opened, and once you close it, you null it out